src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -111,8 +113,8 @@ protected override Task DisposeCoreAsync(bool initialized)
 
         private async Task AddOrRemoveProjectFromStartupProjectList(bool initialize = false)
         {
+            bool isDebuggable = await IsDebuggable().ConfigureAwait(false);
```

Comment:
Wasn't there a concern that not doing this on the UI thread could lead to a race?